network/gossip/libp2p/topology/topicBasedTopology_test.go

```diff
@@ -223,6 +221,168 @@ func (suite *TopicAwareTopologyTestSuite) TestConnectedness_ClusterChannelID() {
+}
+
+// TestLinearFanout_EmptyAllSet evaluates that trying to sample a connected graph when `all`
+// is empty returns an error.
+func (suite *TopicAwareTopologyTestSuite) TestLinearFanout_EmptyAllSet() {
+	// samples 10 all into 'shouldHave'.
+	shouldHave := suite.all.Sample(10)
+
+	// creates a topology for the node
+	top, err := NewTopicBasedTopology(suite.all[0].NodeID, suite.state, suite.subMngr[0])
+	require.NoError(suite.T(), err)
+
+	// sampling with empty `all` and non-empty `shouldHave`
+	_, err = top.sampleConnectedGraph(flow.IdentityList{}, shouldHave)
+	require.Error(suite.T(), err)
+
+	// sampling with empty all and nil `shouldHave`
+	_, err = top.sampleConnectedGraph(flow.IdentityList{}, nil)
+	require.Error(suite.T(), err)
+
+	// sampling with nil all and nil `shouldHave`
+	_, err = top.sampleConnectedGraph(nil, nil)
+	require.Error(suite.T(), err)
+}
+
+// TestConnectedness_Unconditionally evaluates that samples returned by the LinearFanoutGraphSampler with
+// empty `shouldHave` constitute a connected graph.
+func (suite *TopicAwareTopologyTestSuite) TestConnectedness_Unconditionally() {
+	adjMap := make(map[flow.Identifier]flow.IdentityList)
+	for i, id := range suite.all {
+		// creates a topology for the node
+		top, err := NewTopicBasedTopology(id.NodeID, suite.state, suite.subMngr[i])
+		require.NoError(suite.T(), err)
+
+		// samples a graph and stores it in adjacency map
+		sample, err := top.sampleConnectedGraph(suite.all, nil)
+		adjMap[id.NodeID] = sample
+	}
+
+	CheckGraphConnected(suite.T(), adjMap, suite.all, filter.In(suite.all))
+}
+
+// TestConnectedness_Conditionally evaluates that samples returned by the LinearFanoutGraphSampler with
+// some `shouldHave` constitute a connected graph.
+func (suite *TopicAwareTopologyTestSuite) TestConnectedness_Conditionally() {
+	adjMap := make(map[flow.Identifier]flow.IdentityList)
+	for i, id := range suite.all {
+		// creates a topology for the node
+		top, err := NewTopicBasedTopology(id.NodeID, suite.state, suite.subMngr[i])
+		require.NoError(suite.T(), err)
+
+		// samples a graph and stores it in adjacency map
+		// sampling is done with a non-empty `shouldHave` subset of 10 randomly chosen all
+		shouldHave := suite.all.Sample(10)
+		sample, err := top.sampleConnectedGraph(suite.all, shouldHave)
+
+		// evaluates inclusion of should haves in sample
+		for _, shouldHaveID := range shouldHave {
+			require.Contains(suite.T(), sample, shouldHaveID)
+		}
+		adjMap[id.NodeID] = sample
+	}
+
+	CheckGraphConnected(suite.T(), adjMap, suite.all, filter.In(suite.all))
+}
+
+// uniquenessCheck is a test helper method that fails the test if all include any duplicate identity.
+func (suite *TopicAwareTopologyTestSuite) uniquenessCheck(ids flow.IdentityList) {
+	seen := make(map[flow.Identity]struct{})
+	for _, id := range ids {
+		// checks if id is duplicate in ids list
+		_, ok := seen[*id]
+		require.False(suite.T(), ok)
+
+		// marks id as seen
+		seen[*id] = struct{}{}
+	}
```

Comment:
```suggestion
	seen := make(map[flow.Identifier]struct{})
	for _, id := range ids {
		// checks if id is duplicate in ids list
		_, ok := seen[id.NodeID]
		require.False(suite.T(), ok)

		// marks id as seen
		seen[id.NodeID] = struct{}{}
	}
```
Better to use a simpler and smaller key for the map